fsgrid.hpp

```diff
@@ -614,28 +610,147 @@ template <typename T, int stencil> class FsGrid : public FsGridTools{
          this->data = other.getData(); // Copy assignment
       }
 
-      /*! Finalize instead of destructor, as the MPI calls fail after the main program called MPI_Finalize().
-       *  Cleans up the cartesian communicator
+      /*! 
+       *  MPI calls fail after the main program called MPI_Finalize(),
+       *  so this can be used instead of the destructor
+       *  Cleans up the cartesian communicator and datatypes
        */
-      void finalize() {
-         // If not a non-FS process
```

Comment:
Oh, I see better what you mean, I deleted that rank check. Pretty sure rank -1 leaves all that stuff to null so no need for a separate check, imo cleaner to just make sure any unused communicators are null everywhere

Reply:
You can build a test case with the instructions linked to in 
https://github.com/fmihpc/vlasiator/pull/926

(and, this might be a good time to add to the Vlasiator wiki a page about how to do this. Do you think this would be possible for you? Thanks.)

Reply:
Hrm, so is there any indication whether the feature is on?